src/ranged.cpp

```diff
@@ -928,51 +983,69 @@ std::vector<tripoint> game::target( tripoint &p, const tripoint &low, const trip
+                    if( gunmod != NULL ) {
                         mode = gunmod->type_name();
                     }
                 }
-                if (mode != "") {
-                    mvwprintw(w_target, line_number, 14, _("Firing mode: %s"),
-                              mode.c_str());
+                if( mode != "" ) {
+                    mvwprintw( w_target, line_number, 14, _("Firing mode: %s"),
+                               mode.c_str() );
                 }
                 line_number++;
             } else {
                 // throwing something or setting turret's target
-                mvwprintw(w_target, line_number++, 1, _("Range: %d/%d, %s"),
-                          rl_dist(from, p), range, enemiesmsg.c_str());
+                mvwprintw( w_target, line_number++, 1, _("Range: %d/%d, %s"),
+                          rl_dist(from, p), range, enemiesmsg.c_str() );
             }
 
             if( critter != nullptr && u.sees( *critter ) ) {
-                // The 4 is 2 for the border and 2 for aim bars.
-                int available_lines = height - num_instruction_lines - line_number - 4;
+                // The 6 is 2 for the border and 4 for aim bars.
+                int available_lines = compact_window ? 1 :
+                                      ( height - num_instruction_lines - line_number - 6 );
                 line_number = critter->print_info( w_target, line_number, available_lines, 1);
             } else {
                 mvwputch(w_terrain, POSY + p.y - center.y, POSX + p.x - center.x, c_red, '*');
             }
         } else {
-            mvwprintw(w_target, line_number++, 1, _("Range: %d, %s"), range, enemiesmsg.c_str());
+            mvwprintw( w_target, line_number++, 1, _("Range: %d, %s"), range, enemiesmsg.c_str() );
         }
 
         if( mode == TARGET_MODE_FIRE && critter != nullptr && u.sees( *critter ) ) {
-            line_number = u.print_aim_bars( w_target, line_number, relevant, critter );
+            int predicted_recoil = u.recoil;
+            int predicted_delay = 0;
+            if( aim_mode->has_threshold && aim_mode->threshold < u.recoil ) {
+                do{
+                    const int aim_amount = u.aim_per_time( &u.weapon, predicted_recoil );
+                    if( aim_amount > 0 ) {
+                        predicted_delay += 10;
+                        predicted_recoil = std::max( predicted_recoil - aim_amount , 0);
+                    }
+                } while( predicted_recoil > aim_mode->threshold &&
+                          predicted_recoil - sight_dispersion > 0 );
+            } else {
+                predicted_recoil = u.recoil;
+            }
+            line_number = u.print_aim_bars( w_target, line_number, relevant, critter, predicted_recoil );
+            if( aim_mode->has_threshold ) {
+                mvwprintw(w_target, line_number++, 1, _("%s Delay: %i"), aim_mode->name.c_str(), predicted_delay, predicted_recoil );
```

Comment:
Whoops, that's a leftover from some testing. I must have amended the format string but left the vestigial unused arg. I'll get rid of it.